src/custom/hooks/web3.ts

```diff
@@ -0,0 +1,128 @@
+import { Web3Provider } from '@ethersproject/providers'
+import { useWeb3React as useWeb3ReactCore } from '@web3-react/core'
+import { Web3ReactContextInterface } from '@web3-react/core/dist/types'
+import { useEffect, useState, useCallback } from 'react'
+import { isMobile } from 'react-device-detect'
+import { injected, walletconnect, getProviderType, WalletProvider } from 'connectors'
+import { NetworkContextName } from 'constants/misc'
+
+export function useActiveWeb3React(): Web3ReactContextInterface<Web3Provider> {
+  const context = useWeb3ReactCore<Web3Provider>()
+  const contextNetwork = useWeb3ReactCore<Web3Provider>(NetworkContextName)
+  return context.active ? context : contextNetwork
+}
+
+// last wallet provider key used in local storage
+export const STORAGE_KEY_LAST_PROVIDER = 'lastProvider'
+
+export function useEagerConnect() {
+  const { activate, active, connector } = useWeb3ReactCore() // specifically using useWeb3ReactCore because of what this hook does
```

Comment:
funny comment "because of what this hook does"

Reply:
it be that way cause it do